components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorService.java

```diff
@@ -0,0 +1,364 @@
+            throws APIManagementException {
+        AccessTokenInfo accessTokenInfo = new AccessTokenInfo();
+        AccessTokenRequest accessTokenRequest = new AccessTokenRequest();
+        // Get scopes of the application
+        String scopes = getApplicationScopes(appName);
+        if (log.isDebugEnabled()) {
+            log.debug("Set scopes for " + appName + " application using swagger definition.");
+        }
+        // TODO: Get Consumer Key & Secret without creating a new app, from the IS side
+        Map<String, String> consumerKeySecretMap = getConsumerKeySecret(appName);
+        if (log.isDebugEnabled()) {
+            log.debug("Received consumer key & secret for " + appName + " application.");
+        }
+        try {
+            if (grantType.equals(KeyManagerConstants.AUTHORIZATION_CODE_GRANT_TYPE)) {
+                // Access token for authorization code grant type
+                APIMStoreConfigurations storeConfigs = ServiceReferenceHolder.getInstance().getAPIMStoreConfiguration();
+                String callBackURL = storeConfigs.getApimBaseUrl() + "login/callback/" + appName;
+                // Get the Authorization Code
+                if (requestURL.contains("code=")) {
+                    String requestURLQueryParameters = requestURL.split("\\?")[1];
+                    String authorizationCode = requestURLQueryParameters.split("=")[1].split("&")[0];
+                    if (log.isDebugEnabled()) {
+                        log.debug("Authorization Code for the app " + appName + ": " + authorizationCode);
+                    }
+                    // Get Access & Refresh Tokens
+                    accessTokenRequest.setClientId(consumerKeySecretMap.get("CONSUMER_KEY"));
+                    accessTokenRequest.setClientSecret(consumerKeySecretMap.get("CONSUMER_SECRET"));
+                    accessTokenRequest.setGrantType(grantType);
+                    accessTokenRequest.setAuthorizationCode(authorizationCode);
+                    accessTokenRequest.setScopes(scopes);
+                    accessTokenRequest.setCallbackURI(callBackURL);
+                    accessTokenInfo = getKeyManager().getNewAccessToken(accessTokenRequest);
+                } else {
+                    String errorMsg = "No Authorization Code available.";
+                    log.error(errorMsg, ExceptionCodes.ACCESS_TOKEN_GENERATION_FAILED);
+                }
+            } else if (grantType.equals(KeyManagerConstants.PASSWORD_GRANT_TYPE)) {
+                // Access token for password code grant type
+                accessTokenRequest = AuthUtil
+                        .createAccessTokenRequest(userName, password, grantType, refreshToken,
+                                null, validityPeriod, scopes,
+                                consumerKeySecretMap.get("CONSUMER_KEY"), consumerKeySecretMap.get("CONSUMER_SECRET"));
+                accessTokenInfo = getKeyManager().getNewAccessToken(accessTokenRequest);
```

Comment:
since we dont have a else statement is there any possibility that it can effect the follow because accessTokenInfo is getting null?